datumaro/plugins/data_formats/ava/ava.py

```diff
@@ -0,0 +1,222 @@
+# Copyright (C) 2022 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import csv
+import os
+import os.path as osp
+
+import google.protobuf.text_format as text_format
+
+from datumaro.components.annotation import AnnotationType, Bbox, LabelCategories
+from datumaro.components.dataset_base import DatasetItem, SubsetBase
+from datumaro.components.errors import DatasetImportError, MediaTypeError
+from datumaro.components.exporter import Exporter
+from datumaro.components.importer import Importer
+from datumaro.components.media import Image
+from datumaro.util.os_util import find_files
+
+from . import ava_label_pb2
+
+
+class AvaPath:
+    IMAGE_DIR = "frames"
+    IMAGE_EXT = ".jpg"
+    ANNOTATION_DIR = "annotations"
+    ANNOTATION_EXT = ".csv"
+    ANNOTATION_PREFIX = "ava_"
+    ANNOTATION_VERSION = "_v2.2"
+    LABEL_LIST = ANNOTATION_PREFIX + "action_list" + ANNOTATION_VERSION + ".pbtxt"
+    PROPOSAL_EXT = ".pkl"
+
+
+class AvaBase(SubsetBase):
+    def __init__(self, path):
+        if not osp.isfile(path):
+            raise DatasetImportError(f"Can't find JSON file at '{path}'")
+        self._path = path
+
+        subset = (
+            osp.splitext(osp.basename(path))[0]
+            .replace(AvaPath.ANNOTATION_PREFIX, "")
+            .replace(AvaPath.ANNOTATION_VERSION, "")
+        )
+        super().__init__(subset=subset)
+
+        if path.endswith(osp.join(AvaPath.ANNOTATION_DIR, osp.basename(path))):
+            self._rootpath = path.rsplit(AvaPath.ANNOTATION_DIR, maxsplit=1)[0]
+        else:
+            raise DatasetImportError(
+                f"Annotation path ({path}) should be under the directory which is named {AvaPath.ANNOTATION_DIR}. "
+                "If not, Datumaro fails to find the root path for this dataset."
+            )
+
+        if self._rootpath and osp.isdir(osp.join(self._rootpath, AvaPath.IMAGE_DIR)):
+            self._images_dir = osp.join(self._rootpath, AvaPath.IMAGE_DIR)
+        else:
+            raise DatasetImportError(
+                f"Root path ({self._rootpath}) should contain the directory which is named {AvaPath.IMAGE_DIR}. "
+                "If not, Datumaro fails to find the image directory path."
+            )
+
+        self._infos = self._load_infos(osp.dirname(path))
+
+        category_path = osp.join(self._rootpath, AvaPath.ANNOTATION_DIR, AvaPath.LABEL_LIST)
+        self._categories = self._load_categories(category_path)
+
+        self._items = self._load_items(path)
+
+    def _load_infos(self, path):
+        infos = {}
+        for file in os.listdir(path):
+            if file.endswith(AvaPath.PROPOSAL_EXT):
+                name = file.split(".")[0].split("_")[-1]
+                infos[name + "_proposals"] = file
```

Comment:
Since the proposal filename is given by "ava_dense_proposals_train.FAIR.recall_93.9.pkl", I need to first extract only ava_dense_proposals_train, I then choose train by splitting this with "_". Do you have any other suggestion for this?